Momento/Incubating/SimpleCacheClient.cs

```diff
@@ -1,103 +1,559 @@
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using MomentoSdk.Responses;
+using Utils = MomentoSdk.Internal.Utils;
+using MomentoSdk.Incubating.Internal;
 using MomentoSdk.Incubating.Responses;
 
 namespace MomentoSdk.Incubating;
 
-public class SimpleCacheClient : MomentoSdk.SimpleCacheClient
+public class SimpleCacheClient : ISimpleCacheClient
 {
-    public SimpleCacheClient(string authToken, uint defaultTtlSeconds) : base(authToken, defaultTtlSeconds)
+    private readonly ISimpleCacheClient simpleCacheClient;
+    private readonly ScsDataClient dataClient;
+
+    public SimpleCacheClient(ISimpleCacheClient simpleCacheClient, string authToken, uint defaultTtlSeconds, uint? dataClientOperationTimeoutMilliseconds = null)
```

Comment:
not a blocker for this PR, but I would probably expect this constructor to be private, and we would have a public one (or factory method) that handles the construction of the other client implicitly so the user doesn't have to.

Or perhaps you already have a builder somewhere that I haven't seen yet that takes care of this concern.